### PR TITLE
get-topic-query

### PR DESCRIPTION
**Pull request by LW95x**

### CODE REVIEW

File: __tests__/integration.test.js [hunk 1]
```diff
@@ -111,6 +111,33 @@ describe("GET /api/articles", () => {
         });
       });
   });
+  test("200: if a topic query is passed in, should only return topics with that query", () => {
+    return request(app)
+      .get("/api/articles?mitch")
```
Comment (nc-poppy-m) [inline __tests__/integration.test.js]: The syntax isn't quite right here. What if we had multiple queries, like topic and author? This should be `/api/articles?topic=mitch`

File: __tests__/integration.test.js [hunk 2]
```diff
@@ -111,6 +111,33 @@ describe("GET /api/articles", () => {
         });
       });
   });
+  test("200: if a topic query is passed in, should only return topics with that query", () => {
```
Comment (nc-poppy-m) [inline __tests__/integration.test.js:114]: What would we want back if passed a valid, existing topic with no articles yet?

File: db/models/getArticles.model.js [hunk 3]
```diff
@@ -1,11 +1,45 @@
 const db = require("../connection");
 
-exports.findArticles = () => {
-  return db
-    .query(
-      `SELECT 
+exports.findArticles = (topic) => {
+  if (topic) {
```
Comment (nc-poppy-m) [inline db/models/getArticles.model.js]: OPTIONAL(but recommended)  - How could we account for both situations (topic present or not) with a single SQL query and add an extra line of SQL if a topic is present?

File: db/models/getArticles.model.js [hunk 4]
```diff
@@ -1,11 +1,45 @@
...
+    LEFT JOIN comments c ON a.article_id = c.article_id
+    WHERE a.topic = $1
+    GROUP BY a.article_id, a.title, a.topic, a.author, a.created_at, a.votes, a.article_img_url
+    ORDER BY a.created_at DESC`,
+        [topic]
+      )
+      .then(({ rows }) => {
+        const articles = rows[0];
+
+        if (!articles) {
```
Comment (nc-poppy-m) [inline db/models/getArticles.model.js]: Do we want to reject this if there are no articles? What if it's an existing topic with no articles yet?

**Review by nc-poppy-m — CHANGES_REQUESTED** [review]
Hi Liam, there's still a little more to do here:
- You should be testing the endpoint `/api/articles?topic=mitch` not ` /api/articles?mitch`
- What would we want to happen if you were given a valid, existing topic with no articles yet?
_- Optional_: Could you slim your SQL queries down so that you have only one query regardless of whether or not a topic is provided? (see below)

Implement the above and then resubmit this ticket. Nearly there!

**Review by Slyose — APPROVED** [review]
Looks good Li! Good response to the feedback from Poppy. Merge away ✅

**Pull request merged**